src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -574,15 +565,41 @@ private static bool IsOverridable(ISymbol member, INamedTypeSymbol containingTyp
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                if (!member.IsImplicitlyDeclared)
+                // An implicitly declared override is still something the user can provide their own explicit override
+                // for.  This is true for all implicit overrides *except* for the one for `bool object.Equals(object)`.
+                // This override is not one the user is allowed to provide their own override for as it must have a very
+                // particular implementation to ensure proper record equality semantics.
+                if (!member.IsImplicitlyDeclared || IsEqualsObjectOverride(member))
```

Comment:
this is the fix.  we specially case things that aren't truly user overridable due to language rules.

Reply:
I'm a little confused, is overriding `Equals(object)` also invalid for classes too? (That's my overall impression from the PR, despite the title)

Reply:
so, the 'if' checks two things:

First, is the member is implicitly declared.  If *not*, then we consider it.  For normal classes, the member is not implicitly declared, because you have to actually write out your override.

However, with *records* the overrides are implict and are automatically provided.  The user can override all the implicit ones *except* the core Equals(object) one.  

So that's how this check works :)

Reply:
I don't think its a big deal, but the comment could make it a bit more obvious that this is for records, where overriding Equals(object) is a compile error. I'm guessing you know better than me how likely it is that there will be a new language construct added where there is an implicitly declared Equals method that the user _is_ allowed to override :)

Reply:
that's fair :)